model/task/task.go

```diff
@@ -954,12 +967,24 @@ func (t *Task) MarkDependenciesFinished(ctx context.Context, finished bool) erro
 				DependencyTaskIdKey: t.Id,
 			}},
 		},
-		bson.M{
-			"$set": bson.M{bsonutil.GetDottedKeyName(DependsOnKey, "$[elem]", DependencyFinishedKey): finished},
+		[]bson.M{
```

Comment:
We should put some comments in here to explain in words what's happening, since it's a bit complex. Does this change behavior?